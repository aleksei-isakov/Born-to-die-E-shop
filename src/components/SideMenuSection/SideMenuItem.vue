<template>
  <div>
    <div
      v-for="item in items"
      :key="item.id"
      class="side-menu-item"
      @click="onClickClosePopup"
    >
      <base-button-router :path="item.path" class="side-menu-item__title">
        <i :class="`${item.icon} side-menu-item__icon`" />
        {{ item.title }}
      </base-button-router>
    </div>
  </div>
</template>

<script>
import BaseButtonRouter from '@/base_components/BaseButtonRouter/BaseButtonRouter.vue';

export default {
  name: 'SideMenuItem',

  components: { BaseButtonRouter },

  data() {
    return {
      items: [
        {
          title: 'My profile',
          icon: 'fas fa-house-user',
          path: '/profile'
        },

        { title: 'My wishlist', icon: 'fas fa-heart', path: '/wishlist' },

        {
          title: 'My orders',
          icon: 'fas fa-folder',
          path: '/orders'
        },

        {
          title: 'My address book',
          icon: 'fas fa-address-book',
          path: '/address-book'
        }
      ]
    };
  },

  methods: {
    onClickClosePopup() {
      this.$emit('close');
    }
  }
};
</script>

<style lang="scss">
@import '@/scss/CustomVariables.scss';

.side-menu-item {
  display: flex;
  padding: 9px 12px;

  &__icon {
    font-size: 20px;
    color: $primary;
    margin-right: 5px;
  }

  &__title {
    font-size: 16px;
    width: 100%;
    text-align: left;
    color: $primary !important;
  }
}
</style>
