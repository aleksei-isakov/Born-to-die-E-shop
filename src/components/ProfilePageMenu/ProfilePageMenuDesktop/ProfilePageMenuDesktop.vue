<template>
  <side-menu-item class="profile-page-menu__sidebar" color="custom-grey" />
</template>

<script>
import SideMenuItem from '@/components/SideMenuSection/SideMenuItem';

export default {
  name: 'ProfilePageMenuDesktop',

  components: {
    SideMenuItem
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.profile-page-menu {
  &__sidebar {
    background: #ffffff;
    border: 1px solid $light-border-color;
    box-shadow: $shadow;
    height: 450px;
    min-width: 220px;
    padding: 10px;
  }
}
</style>
