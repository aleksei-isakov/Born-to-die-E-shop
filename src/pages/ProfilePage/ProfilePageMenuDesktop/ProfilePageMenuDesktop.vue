<template>
  <div v-if="isDesktop" class="profile-page-menu__container">
    <div class="profile-page-menu__wrapper">
      <div class="profile-page-menu__sections">
        <side-menu-item class="profile-page-menu__sidebar" color="grey" />
        <div class="profile-page-menu__content"></div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import SideMenuItem from '@/components/SideMenuSection/SideMenuItem';

export default {
  name: 'ProfilePageMenuDesktop',

  components: {
    SideMenuItem
  },

  computed: {
    ...mapGetters('ProfilePageModule', ['isDesktop', 'isMobile'])
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.profile-page-menu {
  &__container {
    min-width: 100%;
    min-height: 100%;
  }

  &__wrapper {
    margin: 0 auto;
    width: 70%;
  }

  &__sections {
    display: flex;
    gap: 20px;
    margin-top: 10px;
  }

  &__sidebar,
  &__content {
    background: #ffffff;
    border: 1px solid $light-border-color;
    box-shadow: $shadow;
    height: 450px;
  }

  &__content {
    width: 70%;
  }

  &__sidebar {
    width: 30%;
    min-width: 200px;
    padding: 10px;
  }
}
</style>
