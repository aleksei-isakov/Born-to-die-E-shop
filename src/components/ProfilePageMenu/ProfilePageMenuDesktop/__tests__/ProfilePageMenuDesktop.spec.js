import { shallowMount } from '@vue/test-utils';
import ProfilePageMenuDesktop from '@/components/ProfilePageMenu/ProfilePageMenuDesktop/ProfilePageMenuDesktop';
import SideMenuItem from '@/components/SideMenuSection/SideMenuItem';

let wrapper;

beforeEach(() => {
  wrapper = shallowMount(ProfilePageMenuDesktop, {
    stubs: {
      SideMenuItem
    }
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe('ProfilePageMenuDesktop', () => {
  it('renders the desktop version of ProfilePage', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain sidebar', () => {
    expect(wrapper.findComponent(SideMenuItem).exists()).toBe(true);
  });
});
