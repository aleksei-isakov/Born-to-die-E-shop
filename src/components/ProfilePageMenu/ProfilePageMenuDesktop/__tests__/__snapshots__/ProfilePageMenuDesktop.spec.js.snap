// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePageMenuDesktop renders the desktop version of ProfilePage 1`] = `
<div class="profile-page-menu__sidebar">
  <div class="side-menu-item">
    <base-button-router-stub path="/profile" class="side-menu-item__title side-menu-item__title_custom-grey"><i class="fas fa-house-user side-menu-item__icon side-menu-item__icon_custom-grey"></i>
      My profile
    </base-button-router-stub>
  </div>
  <div class="side-menu-item">
    <base-button-router-stub path="/wishlist" class="side-menu-item__title side-menu-item__title_custom-grey"><i class="fas fa-heart side-menu-item__icon side-menu-item__icon_custom-grey"></i>
      My wishlist
    </base-button-router-stub>
  </div>
  <div class="side-menu-item">
    <base-button-router-stub path="/orders" class="side-menu-item__title side-menu-item__title_custom-grey"><i class="fas fa-folder side-menu-item__icon side-menu-item__icon_custom-grey"></i>
      My orders
    </base-button-router-stub>
  </div>
  <div class="side-menu-item">
    <base-button-router-stub path="/address-book" class="side-menu-item__title side-menu-item__title_custom-grey"><i class="fas fa-address-book side-menu-item__icon side-menu-item__icon_custom-grey"></i>
      My address book
    </base-button-router-stub>
  </div>
</div>
`;
